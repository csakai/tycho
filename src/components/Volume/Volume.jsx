import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

export default class Volume extends React.Component {

    static propTypes = {
        playing: PropTypes.bool,
        onClick: PropTypes.func
    }

    render() {
        return (
            <div
                onClick={this.props.onClick}
                className={cx({
                    'volume': true,
                    'volume volume--playing': this.props.playing,
                    'volume volume--muted': !this.props.playing
                })}>
                <div className="volume__bar"></div>
                <div className="volume__bar"></div>
                <div className="volume__bar"></div>
                <div className="volume__bar"></div>
            </div>
        );
    }
}
