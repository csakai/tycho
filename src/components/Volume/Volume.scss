@import 'globals.scss';

.volume {
    position: relative;
    width: 32px;
    height: 32px;
    display: flex;
    flex-direction: row;
    transform: rotateX(-180deg);

    &__bar {
        width: 8px;
        margin: 0 1px;

        &:nth-of-type(1) {
            height: 32px;
        }

        &:nth-of-type(2) {
            height: 25px;
        }

        &:nth-of-type(3) {
            height: 10px;
        }

        &:nth-of-type(4) {
            height: 18px;
        }

        &::before {
            content: " ";
            height: 100%;
            display: block;
            background-color: $brand-primary;
            pointer-events: none;
            transition: height 1s;
            animation-play-state: paused;
        }
    }

    &--playing:hover {
        .volume__bar {
            &:nth-of-type(1) {
                &::before {
                    animation: animated-bar 1s infinite;
                }
            }

            &:nth-of-type(2) {
                &::before {
                    animation: animated-bar 0.75s infinite;
                }
            }

            &:nth-of-type(3) {
                &::before {
                    animation: animated-bar 0.75s infinite;
                }
            }

            &:nth-of-type(4) {
                &::before {
                    animation: animated-bar 0.5s infinite;
                }
            }
        }
    }

    &--muted {
        .volume__bar {
            &::before {
                height: 4px;
                animation-play-state: paused;
            }
        }
    }
}

@keyframes animated-bar {
    0% {
        height: 100%;
    }
    25% {
        height: 90%;
    }
    50% {
        height: 50%;
    }
    75% {
        height: 70%;
    }
    100% {
        height: 100%;
    }
}
