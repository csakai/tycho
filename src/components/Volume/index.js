import './Volume.css';
export {default} from './Volume';
