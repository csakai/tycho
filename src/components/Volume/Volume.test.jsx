import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Volume from './Volume';

describe('Volume Component', () => {
    describe('render()', () => {
        it('should render the Volume successfully', () => {
            const component = shallow(<Volume />);
            expect(toJson(component)).toMatchSnapshot();
        });
    });
});
