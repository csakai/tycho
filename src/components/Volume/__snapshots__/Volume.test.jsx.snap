// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Volume Component render() should render the Volume successfully 1`] = `
<div
  className="volume volume volume--muted"
>
  <div
    className="volume__bar"
  />
  <div
    className="volume__bar"
  />
  <div
    className="volume__bar"
  />
  <div
    className="volume__bar"
  />
</div>
`;
